.github/workflows/continuous-integration.yaml

```diff
@@ -43,61 +52,111 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-php-
 
-      - name: Disable NPM bridge
-        run: composer config allow-plugins.oat-sa/composer-npm-bridge false --no-interaction --no-plugins
+      - name: Configure NPM bridge plugin
+        run: composer config allow-plugins.oat-sa/composer-npm-bridge $([[ "${{ matrix.node-version }}" ]] && echo "true" || echo "false") --no-interaction --no-plugins
 
       - name: Install dependencies
         run: composer install --no-progress
 
-      - name: Run tests
+      - name: Generate localization files
+        if: ${{ matrix.node-version }}
+        run: php tao/scripts/tools/GenerateTranslationBundles.php
+
+      - name: Front-end tests
+        if: ${{ matrix.node-version }}
+        working-directory: tao/views/build
+        run: |
+          perl -0777 -pi -e "s/(puppeteer[\s\r\n]*:[\s\r\n]*.*args:[\s\r\n]+\[)/\1'--disable-dev-shm-usage','--disable-setuid-sandbox','--single-process',/s" grunt/test.js
```

Comment:
There must be a saner way to set some args, than editing a file on the fly!

Reply:
For sure :)

In fact, I noticed only later that the flags I added probably do not make much of a difference. Let me try to execute the workflow without this ugly line.

Reply:
Yep.

The run was successful without these flags as well. So, while they could be adding to stability of the tests, I guess we can live without this ugly hack.